cloud_utils/vault.py

```diff
@@ -1,136 +1,35 @@
-        )
-
-    return _pod_identity_token
-
-
-def _make_vault_headers(
-    token: Callable[[], Coroutine[Any, Any, str]],
-) -> Callable[[], Coroutine[Any, Any, Dict[str, str]]]:
-    async def _vault_headers() -> Dict[str, str]:
-        return {
-            "X-Vault-Token": await token(),
-        }
-
-    return _vault_headers
+def read_all_secrets(path: str) -> dict[str, str]:
+    client = _get_client(_URL, _TOKEN)
+    read_secrets = client.secrets.kv.v2.read_secret_version(path=path)
+    return read_secrets["data"]["data"]
 
 
-def _make_read_key(
-    base_vault_url: str,
-    headers: Callable[[], Coroutine[Any, Any, dict]],
-):
-    async def read_key(path: str) -> Dict[str, str]:
-        return gamla.pipe(
-            await gamla.get_async_with_headers(
-                (await headers()),
-                30,
-                f"{base_vault_url}/secret/data/{path}",
-            ),
-            lambda response: response.json(),
-            gamla.excepts(
-                KeyError,
-                gamla.make_raise(KeyDoesNotExist),
-                gamla.get_in(["data", "data"]),
-            ),
-        )
+# TODO: do we want to have the ability to read a specific secret?
+# def read_secret(path: str, key: str) -> str:
+#     secrets_data = read_all_secrets(path)
+#     return secrets_data[key]
 
-    return read_key
 
-
-def _make_write_key(
-    base_vault_url: str,
-    headers: Callable[[], Coroutine[Any, Any, dict]],
-):
-    async def write_key(path: str, value: Dict[str, str]):
-        return await gamla.post_json_with_extra_headers_async(
-            (await headers()),
-            30,
-            f"{base_vault_url}/secret/data/{path}",
-            {"data": value},
-        )
-
-    return write_key
-
-
-def _make_update_key(read_key: Callable, write_key: Callable):
-    async def update_key(path: str, value: Dict[str, str]):
-        current_value = {}
-        try:
-            current_value = await read_key(path)
-        except KeyDoesNotExist:
-            pass
-        return await write_key(path, gamla.merge(current_value, value))
-
-    return update_key
-
-
-def make_vault(host: str, role: str, token: Optional[str]):
-    async def get_token():
-        return token
-
-    base_vault_url = f"{host}/v1"
-    headers = _make_vault_headers(
-        get_token if token else _make_pod_identity_token(role, base_vault_url),
+def write_or_update_secrets(path: str, secrets: dict[str, str], mount_point: str):
+    client = _get_client(_URL, _TOKEN)
+    secrets_data = read_all_secrets(path)
+    secrets_data.update(secrets)
+    client.secrets.kv.v2.create_or_update_secret(
+        path=path,
+        secret=secrets_data,
+        mount_point=mount_point,
     )
 
-    write_key, read_key = _make_write_key(base_vault_url, headers), _make_read_key(
-        base_vault_url,
-        headers,
-    )
-    return write_key, read_key, _make_update_key(read_key, write_key)
+
+def make_vault():
```

Comment:
make_vault function still needs to get as input `host: str, role: str, token: Optional[str]`

What you can do here is instead of initializing the vault client every time you call either write_or_update_secrets or read_all_secerts is to initialize it once in make_vault and then pass on the client to each one of these functions (similar to what they did in the previous code)

Reply:
Regarding my comment about initializing the client once here and passing it to the functions, you will have something similar to:

```
def make_vault(host: str, role: str, token: Optional[str]):
     client = _get_client(host, token)
     write_or_update_secrets, read_all_secrets = _write_or_update_secrets(client), _read_all_secerts(client)

    return write_or_update_secrets, read_all_secrets 
```